### PR TITLE
Stateless optimisers re-done

### PR DESCRIPTION
**Pull request by 0xZenus**
# Title

Integration to stateless optimisers.

## Description

- duplicates the existing Optimiser flow and adjusts the main version to the stateless SDK (controlled by REACT_APP_STATELESS_SDK)
- stores the information in redux

## Amplify hosted environment link

https://feat-stateless-optimisers-latest.d2f2fhja7tsifl.amplifyapp.com/

### CODE REVIEW

File: src/app/features/stateless-optimisers/thunks.ts [hunk 1]
```diff
@@ -0,0 +1,73 @@
+import { createAsyncThunk } from '@reduxjs/toolkit';
+import { depositAndRegister, getAllMellowProducts } from '@voltz-protocol/v1-sdk';
+import { ethers } from 'ethers';
+
+import { OptimiserInfo } from './types';
+
+// TODO: enrich this mapping
+const mapRouter = (r: Awaited<ReturnType<typeof getAllMellowProducts>>[0]): OptimiserInfo => r;
```
Comment (ssbarbee) [inline src/app/features/stateless-optimisers/thunks.ts:8]: can we remove this

File: src/app/features/stateless-optimisers/thunks.ts [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+      const response = await depositAndRegister({
+        optimiserId,
+        amount,
+        spareWeights,
+        registration,
+        signer,
+      });
+
+      if (!response.newOptimiserState) {
+        throw new Error('New state not returned');
```
Comment (ssbarbee) [inline src/app/features/stateless-optimisers/thunks.ts:65]: do   return rejectThunkWithError(thunkAPI, new Error('New state not returned'));

File: src/routes/LPOptimisers/VaultFormRoute/VaultWithdrawRolloverForm/VaultWithdrawRolloverForm.tsx [hunk 3]
```diff
@@ -37,12 +41,25 @@ export const VaultWithdrawRolloverForm: React.FunctionComponent<VaultWithdrawRol
   const weights = distribution === 'automatic' ? automaticWeights : manualWeights;
   const combinedWeightValue = weights.reduce((total, weight) => total + weight.distribution, 0);
 
+  const spareWeights = weights
```
Comment (ssbarbee) [inline src/routes/LPOptimisers/VaultFormRoute/VaultWithdrawRolloverForm/VaultWithdrawRolloverForm.tsx:44]: lets extract utility from this. Maybe inside helpers?

File: src/routes/LPOptimisers/useLPVaults.ts [hunk 4]
```diff
@@ -1,53 +1,25 @@
...
+export const useLPVaults = () => {
+  const { signer } = useWallet();
 
-  useEffect(() => {
-    if (!shouldInitVaults) {
-      return;
-    }
-    void dispatch(initialiseVaultsThunk());
-  }, [shouldInitVaults]);
+  const vaultsLoaded = useAppSelector((state) => state.statelessOptimisers.optimisersLoadedState);
```
Comment (ssbarbee) [inline src/routes/LPOptimisers/useLPVaults.ts:13]: can we create a selector for this? same as selectOptimisers

File: src/routes/LPOptimisers/VaultFormRoute/VaultWithdrawRolloverForm/VaultWithdrawRolloverForm.tsx [hunk 5]
```diff
@@ -37,12 +41,25 @@ export const VaultWithdrawRolloverForm: React.FunctionComponent<VaultWithdrawRol
...
+
+    if (!vault.vaults[vaultIndex].withdrawable) {
       return;
     }
+
     setWithdrawOrRolloverState(WithdrawStates.WITHDRAW_PENDING);
-    void vault.withdraw(vaultIndex).then(
+    void executeWithdraw({
+      optimiserId: vault.optimiserId,
+      vaultId: vault.vaults[vaultIndex].vaultId,
```
Comment (ssbarbee) [inline src/routes/LPOptimisers/VaultFormRoute/VaultWithdrawRolloverForm/VaultWithdrawRolloverForm.tsx:60]: perhaps vault = vault.vaults[vaultIndex] on top would make for less changes across code

**Review by ssbarbee — APPROVED** [review]

File: src/app/features/stateless-optimisers/thunks.ts (hunk 2 above)
Comment (0xZenus) [reply src/app/features/stateless-optimisers/thunks.ts]: e61fdf7d222505075152d9b57861c4486326952e

File: src/routes/LPOptimisers/VaultFormRoute/VaultWithdrawRolloverForm/VaultWithdrawRolloverForm.tsx (hunk 3 above)
Comment (0xZenus) [reply src/routes/LPOptimisers/VaultFormRoute/VaultWithdrawRolloverForm/VaultWithdrawRolloverForm.tsx]: dc6a8d30ed67af8aba91cf3ea2a8fcdd1bb73dab

File: src/routes/LPOptimisers/useLPVaults.ts (hunk 4 above)
Comment (0xZenus) [reply src/routes/LPOptimisers/useLPVaults.ts]: 8584ee129395fff176ac73fc3ca7ff3e2c830a4b

File: src/routes/LPOptimisers/VaultFormRoute/VaultWithdrawRolloverForm/VaultWithdrawRolloverForm.tsx (hunk 5 above)
Comment (0xZenus) [reply src/routes/LPOptimisers/VaultFormRoute/VaultWithdrawRolloverForm/VaultWithdrawRolloverForm.tsx]: 96a6f6043cbbb68b1f36d330e8cecdd5ce4c5384

**Pull request merged**